### PR TITLE
added rl config (Use RL for model training #586)

### PR DESCRIPTION
**Pull request by fahuddin**
# [Feat] Add Reinforcement Learning Configuration Options for Classifier Training

## FIX

*Link to related issue if applicable*

## Summary

This PR adds foundational support for Reinforcement Learning (RL) options in classifier model training. The implementation introduces a configuration schema and parsing infrastructure that allows RL-based training to be toggled and configured from `config/config.yaml` without disrupting the existing supervised LoRA training pipeline.

#### `candle-binding/src/core/config_loader.rs`
- Added `RLConfig` struct with fields:
  - `enabled`: Toggle RL training on/off
  - `algorithm`: Algorithm selection (e.g., "ppo", "a2c", "dqn")
  - `learning_rate`: Learning rate for RL policy updates (default: 1e-5)
  - `gamma`: Discount factor for reward accumulation (default: 0.99)
  - `batch_size`: Batch size for RL training (default: 16)
  - `update_epochs`: Number of policy update epochs per rollout (default: 4)
  - `reward_metric`: Metric to compute reward signals (e.g., "accuracy", "f1", default: "accuracy")

- Added `GlobalConfigLoader::load_classifier_rl_config()` method to parse RL options from `config/config.yaml` under `classifier.rl_training` key
- Added `GlobalConfigLoader::load_classifier_rl_config_safe()` safe wrapper with sensible defaults fallback
- All parsing uses existing YAML hierarchical path extraction for consistency with other config loaders

#### `config/config.yaml`
- Added `classifier.rl_training` block with documented defaults:
  ```yaml
  classifier:
    rl_training:
      enabled: false            # RL training toggle
      algorithm: "ppo"        # Algorithm choice
      learning_rate: 1e-05
      gamma: 0.99
      batch_size: 16
      update_epochs: 4
      reward_metric: "accuracy"

### CODE REVIEW

**Comment by Xunzhuo** [comment]
@fahuddin can you explain what you are trying to do in this PR?

**Comment by fahuddin** [comment]
Hi. Based on issue **Use RL for model training #586**, I just set up a RLConfig struct and added new fields to accomdate for RL training. Please correct me if I'm wrong.
